examples/tree_1d_dgsem/elixir_euler_quasi_1d_discontinuous.jl

```diff
@@ -0,0 +1,85 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the quasi 1d compressible Euler equations
+# See Chan et al.  https://doi.org/10.48550/arXiv.2307.12089 for details
+
+equations = CompressibleEulerEquationsQuasi1D(1.4)
+
+"""
+    initial_condition_discontinuity(x, t, equations::CompressibleEulerEquations1D)
+
+A discontinuous initial condition taken from
+- Jesse Chan, Khemraj Shukla, Xinhui Wu, Ruofeng Liu, Prani Nalluri (2023)
+    High order entropy stable schemes for the quasi-one-dimensional
+    shallow water and compressible Euler equations
+    [DOI: 10.48550/arXiv.2307.12089](https://doi.org/10.48550/arXiv.2307.12089)   
+"""
+function initial_condition_discontinuity(x, t,
+                                         equations::CompressibleEulerEquationsQuasi1D)
+    rho = (x[1] < 0) ? 3.4718 : 2.0
+    v1 = (x[1] < 0) ? -2.5923 : -3.0
+    p = (x[1] < 0) ? 5.7118 : 2.639
+    a = (x[1] < 0) ? 1.0 : 1.5
+
+    return prim2cons(SVector(rho, v1, p, a), equations)
+end
+
+initial_condition = initial_condition_discontinuity
+
+volume_flux = (flux_chan_etal, flux_nonconservative_chan_etal)
+surface_flux = (flux_lax_friedrichs, flux_nonconservative_chan_etal)
```

Comment:
Very minor sidenote: Usually we have the surface flux first, then the volume flux/integral.